src/connector/src/sink/snowflake_redshift/redshift.rs

```diff
@@ -0,0 +1,921 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::num::NonZero;
+use std::fmt::Write;
+use std::time::Duration;
+
+use anyhow::anyhow;
+use bytes::BytesMut;
+use phf::{Set, phf_set};
+use risingwave_common::array::StreamChunk;
+use risingwave_common::catalog::{ColumnDesc, ColumnId, Field, Schema};
+use risingwave_common::types::DataType;
+use risingwave_pb::connector_service::sink_metadata::SerializedMetadata;
+use risingwave_pb::connector_service::{SinkMetadata, sink_metadata};
+use sea_orm::DatabaseConnection;
+use serde::Deserialize;
+use serde_json::json;
+use serde_with::{DisplayFromStr, serde_as};
+use thiserror_ext::AsReport;
+use tokio::sync::mpsc::{UnboundedSender, unbounded_channel};
+use tokio::time::{MissedTickBehavior, interval};
+use tonic::async_trait;
+use tracing::warn;
+use with_options::WithOptions;
+
+use crate::connector_common::IcebergSinkCompactionUpdate;
+use crate::enforce_secret::EnforceSecret;
+use crate::sink::coordinate::CoordinatedLogSinker;
+use crate::sink::file_sink::opendal_sink::FileSink;
+use crate::sink::file_sink::s3::{S3Common, S3Sink};
+use crate::sink::jdbc_jni_client::{self, JdbcJniClient};
+use crate::sink::remote::CoordinatedRemoteSinkWriter;
+use crate::sink::snowflake_redshift::{
+    __OP, __ROW_ID, AugmentedChunk, SnowflakeRedshiftSinkS3Writer, build_opendal_writer_path,
+};
+use crate::sink::writer::SinkWriter;
+use crate::sink::{
+    Result, Sink, SinkCommitCoordinator, SinkCommittedEpochSubscriber, SinkError, SinkParam,
+    SinkWriterMetrics,
+};
+
+pub const REDSHIFT_SINK: &str = "redshift";
+
+fn build_full_table_name(schema_name: Option<&str>, table_name: &str) -> String {
+    if let Some(schema_name) = schema_name {
+        format!(r#""{}"."{}""#, schema_name, table_name)
+    } else {
+        format!(r#""{}""#, table_name)
+    }
+}
+
+fn build_alter_add_column_sql(
+    schema_name: Option<&str>,
+    table_name: &str,
+    columns: &Vec<(String, String)>,
+) -> String {
+    let full_table_name = build_full_table_name(schema_name, table_name);
+    // redshift does not support add column IF NOT EXISTS yet.
+    jdbc_jni_client::build_alter_add_column_sql(&full_table_name, columns, false)
+}
+
+#[serde_as]
+#[derive(Debug, Clone, Deserialize, WithOptions)]
+pub struct RedShiftConfig {
+    #[serde(rename = "jdbc.url")]
+    pub jdbc_url: String,
+
+    #[serde(rename = "user")]
+    pub username: Option<String>,
+
+    #[serde(rename = "password")]
+    pub password: Option<String>,
+
+    #[serde(rename = "schema")]
+    pub schema: Option<String>,
+
+    #[serde(rename = "target.table.name")]
```

Comment:
Since iceberg call this `table.name`, we can make it consistent here and call it `table.name` as well.